AlphaWallet/Market/UniversalLinkHandler.swift

```diff
@@ -56,7 +63,16 @@ public class UniversalLinkHandler {
     //link has shortened price and expiry and must be expanded
     func parseUniversalLink(url: String) -> SignedOrder? {
         let linkInfo = b64SafeEncodingToRegularEncoding(url.substring(from: urlPrefix.count))
-        guard let linkBytes = Data(base64Encoded: linkInfo)?.array else { return nil }
+        guard var linkBytes = Data(base64Encoded: linkInfo)?.array else { return nil }
+        let encodingByte = linkBytes[0]
+        switch encodingByte {
+        case Constants.oldFormat: break
+        //new link format, remove extra byte and continue
+        case Constants.notSpawnable: linkBytes.remove(at: 0)
+        case Constants.spawnable: return handleSpawnableLink(linkBytes: linkBytes)
+        case Constants.customizable: return handleSpawnableLink(linkBytes: linkBytes)
+        default: break
+        }
```

Comment:
A `handleNonSpawnableLink()` would be good?

Reply:
We can encode the link format byte value into the `LinkFormat` enum, keeping the byte values together in the enum makes it a bit easier to maintain them (and remove the corresponding `oldFormat` in `Constants.swift`):

    enum LinkFormat: UInt8 {
        case unassigned = 0x00
        case normal = 0x01
        case spawnable = 0x02
        case customizable = 0x03
    }

Then we can replace the switch with:

    if let format = LinkFormat(rawValue: encodingByte) {
        switch format {
        case .unassigned:
            break
        case .normal:
            //new link format, remove extra byte and continue
            linkBytes.remove(at: 0)
        case .spawnable:
            return handleSpawnableLink(linkBytes: linkBytes)
        case .customizable:
            return handleSpawnableLink(linkBytes: linkBytes)
        }
    } else {
        //TODO invalid format. Do something
    }